src/plugins/ui_feature/autofill.ts

```diff
@@ -238,6 +246,108 @@ export class AutofillPlugin extends UIPlugin {
     }
   }
 
+  private collectBordersData(
+    data: Omit<AutoFillCellCommand, "type">,
+    borders: Record<UID, Border>,
+    bordersPositions: Record<number, Zone[]>
+  ) {
+    const id = getItemId(data.border, borders);
+    if (!(id in bordersPositions)) {
+      bordersPositions[id] = [];
+    }
+    bordersPositions[id].push(positionToZone({ col: data.col, row: data.row }));
+  }
+
+  private collectConditionalFormatsData(
+    data: Omit<AutoFillCellCommand, "type">,
+    cfNewRanges: Record<UID, string[]>
+  ) {
+    const { originCol, originRow, col, row } = data;
+    const sheetId = this.getters.getActiveSheetId();
+    const cfOrigin = this.getters.getRulesByCell(sheetId, originCol, originRow);
+    const xc = toXC(col, row);
+    for (const cf of cfOrigin) {
+      if (!(cf.id in cfNewRanges)) {
+        cfNewRanges[cf.id] = [];
+      }
+      cfNewRanges[cf.id].push(xc);
+    }
+  }
+
+  private collectDataValidationsData(
+    data: Omit<AutoFillCellCommand, "type">,
+    dvNewZones: Record<UID, Zone[]>
+  ) {
+    const { originCol, originRow, col, row } = data;
+    const sheetId = this.getters.getActiveSheetId();
+    const cellPosition = { sheetId, col: originCol, row: originRow };
+    const dvOrigin = this.getters.getValidationRuleForCell(cellPosition);
+    if (!dvOrigin) {
+      return;
+    }
+    if (!(dvOrigin.id in dvNewZones)) {
+      dvNewZones[dvOrigin.id] = [];
+    }
+    dvNewZones[dvOrigin.id].push(positionToZone({ col, row }));
+  }
+
+  private autofillCell(data: Omit<AutoFillCellCommand, "type">) {
+    this.dispatch("UPDATE_CELL", {
+      sheetId: this.getters.getActiveSheetId(),
+      col: data.col,
+      row: data.row,
+      content: data.content || "",
+      style: data.style || null,
+      format: data.format || "",
+    });
+    this.dispatch("AUTOFILL_CELL", data);
```

Comment:
Still needed for Odoo in master, and it's not an issue it's a UI command

Reply:
maybe add this as a comment in the code, so we don't forget ?